src/main/java/com/jfrog/ide/idea/scan/SourceCodeScannerManager.java

```diff
@@ -131,13 +148,37 @@ public void onThrowable(@NotNull Throwable error) {
         executor.submit(createRunnable(sourceCodeScanTask, latch, log));
     }
 
-    private void sourceCodeScanAndUpdate(ProgressIndicator indicator, Runnable checkCanceled, Logger log) {
+    private void sourceCodeScanAndUpdate(ProgressIndicator indicator, Runnable checkCanceled, Logger log) throws IOException {
         indicator.setText("Running advanced source code scanning");
+        JFrogApplicationsConfig projectConfig = parseJFrogApplicationsConfig();
+
+        if (projectConfig != null) {
+            for (ModuleConfig moduleConfig : projectConfig.getModules()) {
+                scan(moduleConfig, indicator, checkCanceled, log);
+            }
+        } else {
+            scan(null, indicator, checkCanceled, log);
+        }
+    }
+
+    private void scan(ModuleConfig moduleConfig, ProgressIndicator indicator, Runnable checkCanceled, Logger log) {
         double fraction = 0;
-        for (ScanBinaryExecutor scanner : scanners) {
+        for (SourceCodeScanType scannerType : scanners.keySet()) {
             checkCanceled.run();
+            ScanBinaryExecutor scanner = scanners.get(scannerType);
+            ScannerConfig scannerConfig = null;
+            if (moduleConfig != null) {
+                //  If requested skip the scanner.
```

Comment:
//  If requested skip the scanner.
-->
 // Skip the scanner If requested.

(notice the redundant white-space I removed above).